Building Microservice Natively: Added new files for this section.

scottrifenbark opened the pull request:
The previous section "Host Configuration" was replaced by a
new section called "Building Microservices Natively".  The
files are still living under agl-documentation/host-configuration/
docs.  But, they are all named differently.

Signed-off-by: Scott Rifenbark <srifenbark@gmail.com>

jobol commented:
I have 2 remarks:
 - libmicrohttpd should now be at least 0.9.60
 - the way of using template is probably out-dated

I'm adding other reviewers (or trying to..)

scottrifenbark commented:
I updated the version for "libmirohttpd" to 0.9.60.  I need some other expert to look at the way for using the template.

scottrifenbark commented:
I have addressed the comments for this PR.  However, I am going to close the PR and re-open a new one that comes from my fork and uses a consistent branch name of "sandbox/scottrif29/scottdocs".
